nova/src/folding/hypernova/cyclefold/nimfs/mod.rs

```diff
@@ -0,0 +1,304 @@
+#![allow(unused)]
+
+use ark_crypto_primitives::sponge::{Absorb, CryptographicSponge};
+use ark_ec::short_weierstrass::{Projective, SWCurveConfig};
+use ark_ff::PrimeField;
+use ark_std::marker::PhantomData;
+
+use ark_spartan::polycommitments::{PolyCommitmentScheme, PolyCommitmentTrait};
+
+use crate::commitment::{Commitment, CommitmentScheme};
+
+use crate::folding::hypernova::nimfs::NIMFSProof as HNProof;
+pub use crate::folding::hypernova::nimfs::SQUEEZE_ELEMENTS_BIT_SIZE;
+
+use super::{secondary, Error};
+use crate::{absorb::CryptographicSpongeExt, r1cs, utils::cast_field_element_unique};
+
+pub(crate) use crate::folding::cyclefold::{
+    CCSInstance, CCSShape, CCSWitness, LCCSInstance, R1CSShape, RelaxedR1CSInstance,
+    RelaxedR1CSWitness,
+};
+
+/// Non-interactive multi-folding scheme proof.
+pub struct NIMFSProof<
+    G1: SWCurveConfig,
+    G2: SWCurveConfig,
+    C1: PolyCommitmentScheme<Projective<G1>>,
+    C2: CommitmentScheme<Projective<G2>>,
+    RO,
+> {
+    pub(crate) commitment_T: C2::Commitment,
+    pub(crate) commitment_W_proof: secondary::Proof<G2, C2>,
+    pub(crate) hypernova_proof: HNProof<Projective<G1>, RO>,
+    _poly_commitment: PhantomData<C1::Commitment>,
+}
+
+impl<G1, G2, C1, C2, RO> NIMFSProof<G1, G2, C1, C2, RO>
+where
+    G1: SWCurveConfig<BaseField = G2::ScalarField, ScalarField = G2::BaseField>,
+    G2: SWCurveConfig,
+    C1: PolyCommitmentScheme<Projective<G1>>,
+    C2: CommitmentScheme<Projective<G2>>,
+    G1::BaseField: PrimeField + Absorb,
+    G2::BaseField: PrimeField + Absorb,
+    RO: CryptographicSponge,
+{
+    pub fn prove(
+        _ck: &C1::PolyCommitmentKey,
```

Comment:
why is this here? there shouldn't be any need to compute polynomial commitments anywhere. if we're going to hash the key as part of initializing the random oracle (never a bad idea, but i'm not certain whether it's necessary), we should use the `EvalVerifierKey` instead of the `PolyCommitmentKey` (the latter is huge).

Reply:
Ah sorry, left this in when things were in progress and wasn't sure if I'd need it in some way. Thanks for the catch, and will remove.